google/cloud/aiplatform/jobs.py

```diff
@@ -15,10 +15,214 @@
 # limitations under the License.
 #
 
+from abc import abstractclassmethod
+from typing import Iterable, Optional, Union
 
-class Job:
+from google.cloud import storage
+from google.cloud import bigquery
+
+from google.auth import credentials as auth_credentials
+
+from google.cloud.aiplatform import base
+from google.cloud.aiplatform.gapic import JobState
+from google.cloud.aiplatform.gapic import JobServiceClient
+from google.cloud.aiplatform.utils import full_resource_name
+
+
+class _Job(base.AiPlatformResourceNoun):
+    """
+    Class that represents a general Job resource in AI Platform (Unified).
+    Cannot be directly instantiated.
+
+    Serves as base class to specific Job types, i.e. BatchPredictionJob or
+    DataLabelingJob to re-use shared functionality.
+
+    Subclasses requires one class attribute:
+
+    getter_method (str): The name of JobServiceClient getter method for specific
+    Job type, i.e. 'get_custom_job' for CustomJob
+    """
+
+    client_class = JobServiceClient
+    _is_client_prediction_client = False
+
+    @property
+    @abstractclassmethod
+    def getter_method(cls) -> str:
```

Comment:
Maybe this should be private.

Reply:
Done, also changed `self._job_subclass_getter_method`